src/sqs/backfill.ts

```diff
@@ -20,6 +25,14 @@ export const backfillQueueMessageHandler: MessageHandler<BackfillMessagePayload>
 	});
 
 	const backfillData = { ...context.payload };
+	const rateLimitResponse = (await getRateRateLimitStatus(backfillData, context.log))?.data;
```

Comment:
Just for code clarity, do you think combining those two lines is easier to read? Like:

```javascript

  // const rateLimitResponse = (await getRateRateLimitStatus(backfillData, context.log))?.data;
	// Check if the rate limit is exceeding self-imposed limit
	if (await isRateLimitExceedingSoftLimit(backfillData, context.log, jiraHost)) {
		context.log.info("Rate limit internal threshold exceeded, delaying backfilling message.");
		return await sqsQueues.backfill.changeVisibilityTimeout(context.message, getRateResetTime(rateLimitResponse), context.log);
	}



const isRateLimitExceedingSoftLimit = async (backfillData: BackfillMessagePayload, logger: Logger) : Promise<boolean> => {
  const rateLimitResponse = (await getRateRateLimitStatus(backfillData, context.log))?.data;
	const threshold = await numberFlag(NumberFlags.PREEMPTIVE_RATE_LIMIT_THRESHOLD, 100, jiraHost);
	const { core, graphql } = rateLimitResponse.resources;
  ...
}
```

Reply:
I agree it would be nice, i kept it out because we re-use the response object to get the reset times.

Since it forces an early return it needed something in that scope........... I could make the whole thing part of the if and if it returns true then return... hmmmmmm


....
// change to the existing handler function
if (await checkPreemptiveRateLimitBlahBlahSomethingSomething(backfillData, context.log, jiraHost)) {
		context.log.info("Rate limit internal threshold exceeded, delaying backfilling message.");
		return;
	}
...

Then the checkPreemptiveRateLimitBlahBlahSomethingSomething function will
- test rate limit v threshold
- return false if she's all good
- delay the message and return true of threshold breached

It makes that core function a lot tidier and isolates all the preemptive stuff to a nice little bubble in the corner?